### PR TITLE
Added a real-time debug UI

### PR DESCRIPTION
**Pull request by eko**
This debug UI leverages on Server-Sent Events to bring a real-time view of documents activity in collection.

debug-ui

### CODE REVIEW

File: internal/service/debug.go [hunk 1]
```diff
@@ -0,0 +1,13 @@
+package service
+
+import (
+	"github.com/etf1/kafka-mongo-watcher/internal/debug"
+)
+
+func (container *Container) GetDebugger() *debug.Debugger {
+	if container.debugger == nil {
```
Comment (iotafr) [inline internal/service/debug.go:8]: Maybe we can just instantiate the debugger if `container.Cfg.HttpServer.DebugEnabled == true` AND return `nil` if it's not the case.

File: internal/http/server.go [hunk 2]
```diff
@@ -50,18 +59,38 @@ func (s *TechServer) Start(ctx context.Context) error {
...
+	debugger Debugger,
+) http.Handler {
 	livenessHandler := handler.NewLiveness(logger)
 
 	router := mux.NewRouter()
 	router.Methods(http.MethodGet).Path("/metrics").Handler(promhttp.Handler())
 	router.Methods(http.MethodGet).Path("/liveness").Handler(livenessHandler)
 	router.Methods(http.MethodGet).Path("/readiness").Handler(livenessHandler)
 
+	if debugEnabled {
```
Comment (iotafr) [inline internal/http/server.go:79]: If think it is unnecessary to have a boolean here. Just check if the debugger instance is not nil.

File: public/src/component/App.js [hunk 3]
```diff
@@ -0,0 +1,76 @@
...
+        const filteredEvents = events.filter(event => event !== null).slice(-500);
+        setEvents([...filteredEvents, event]);
+
+        if (automaticScrollEnabled) {
+            window.scrollTo(0, document.body.scrollHeight);
+        }
+    }, [event]);
+
+    return html`
+        <${Header} />
```
Comment (iotafr) [inline public/src/component/App.js:34]: Maybe you can add some information about the database and the name of the collection currently watched.

File: internal/service/kafka.go [hunk 4]
```diff
@@ -60,6 +64,10 @@ func (container *Container) decorateKafkaClientWithOpenTelemetry(producer *kafka
 	)
 }
 
+func (container *Container) decorateKafkaClientWithDebugger(client kafka.Client) kafka.Client {
```
Comment (iotafr) [inline internal/service/kafka.go:67]: It think you can only decorate the client if `container.Cfg.HttpServer.DebugEnabled == true` and always apply the decorator to the client.

**Review by iotafr — APPROVED** [review]

**Comment by iotafr** [comment]
Great job ! 👍

**Comment by eko** [comment]
@iotafr Thanks for your review, I just updated the code and also updated the debugger / UI to add some context informations such as the database name and the collection name (we could add more later if needed).

Here is the UI up-to-date:

Capture d’écran 2022-02-15 à 13 50 36

**Pull request merged**